zip-liberated-payments.rst

```diff
@@ -0,0 +1,410 @@
+
+Zcash protocol terms are as defined in the Zcash Protocol Specification. [#protocol]_
+
+.. _applink:
+By “applink” we mean a platform mechanism for triggering local applications using HTTP links, such as `App Links`_ on Android, `Universal Links`_ on iOS, or `App URI handlers`_ on Windows.
+
+
+Abstract
+========
+
+This proposal defines a mechanism for sending a Zcash payment encapsulated in a URI string. This enables sending Zcash funds over any secure channel, such as via a messaging app, even if the recipient does not yet have Zcash software installed and does not have a Zcash payment address. This is implemented by having the URI convey the secret spending key of an ephemeral Zcash “wallet address”, to which the funds have been transferred. Anyone who learns the URI can accept this payment, by a “finalization” process which uses the key given in the URI to transfer the encapsulated funds to their own wallet. After the payment is finalized, via a suitable on-chain transaction by the recipient, it becomes irrevocable.
+
+The proposal specifies the syntax and semantics of Payment-Encapsulating URIs, the workflow of generating and handling them, and requisite infrastructure.
+
+At its core, a URI encapsulated payment communicates the existence of a transaction (specifically a note committing to an amount of funds) to a receiving client.  The URI encode contains the amount of the payment and a key used to derive all necessary randomness for note construction including the address and secret key needed to spend it.
+
+Usage Story
+-----------
+
+Alice wants to send a 1.23 ZEC payment to Bob. She generates a fresh ephemeral spending key and sends 1.2301 ZEC to it. This transaction is submitted to the network and mined. Meanwhile, Alice constructs a URI containing the ephemeral spending key and appropriate metadata, and sends it to Bob over an end-to-end encrypted channel such as Signal or WhatsApp.
+
+Bob receives a message, which looks as follows:
+
+    This message contains a Zcash payment.
+    Click the following link to view it and receive the funds using your Zcash wallet app: 
+    
+https://pay.withzcash.com:65536/v1#amount=1.23&desc=Payment+for+foo&key=...
+    If you do not yet have a Zcash wallet app, see: https://z.cash/wallets
+
+Bob clicks the link. His Zcash mobile wallet app (which is already installed and has configured itself as a handler for URLs of that form) shows up, and tells Bob that a payment of 1.23 ZEC awaits him. The wallet app confirms the existence of the pertinent transactions on the blockchain, and then offers to finalize the payment. Bob clicks the “Finalize” button, and his wallet app generates a transaction moving the money to his own address (using the extra 0.0001 ZEC he has received to pay the transaction fee). When this transaction is confirmed on-chain, Alice’s and Bob’s wallets both indicate that the payment is finalized, and thus Bob can send the funds to other parties.
```

Comment:
Bob _may_ have a wallet installed, but may need to do as the message suggests, which is to install and set up a wallet first.

Reply:
Yeah. But I'm not sure how to explicitly cover both scenarios in this simple Use Case, without hampering readability. So I was counting on that message text as an adequate cue, while focusing on the common flow. Suggestions welcome!